src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -300,6 +300,31 @@ impl TypedNativeFunction {
                          TypeSignature::UIntType,
                          TypeSignature::IntType],
                     BUFF_32.clone()))),
+            StxTransfer =>
+                Simple(SimpleNativeFunction(FunctionType::Fixed(FixedFunction {
+                    args: vec![
+                        FunctionArg::new(TypeSignature::UIntType,
+                                         ClarityName::try_from("amount".to_owned())
+                                         .expect("FAIL: ClarityName failed to accept default arg name")),
+                        FunctionArg::new(TypeSignature::PrincipalType,
+                                         ClarityName::try_from("sender".to_owned())
+                                         .expect("FAIL: ClarityName failed to accept default arg name")),
+                        FunctionArg::new(TypeSignature::PrincipalType,
+                                         ClarityName::try_from("recipient".to_owned())
+                                         .expect("FAIL: ClarityName failed to accept default arg name")),
+                    ],
+                    returns: TypeSignature::new_response(TypeSignature::BoolType, TypeSignature::UIntType) }))),
+            StxBurn =>
+                Simple(SimpleNativeFunction(FunctionType::Fixed(FixedFunction {
+                    args: vec![
+                        FunctionArg::new(TypeSignature::UIntType,
+                                         ClarityName::try_from("amount".to_owned())
+                                         .expect("FAIL: ClarityName failed to accept default arg name")),
+                        FunctionArg::new(TypeSignature::PrincipalType,
```

Comment:
Same concern here -- doesn't this type definition allow anyone to burn anyone's STX?